drgn/helpers/linux/fs.py

```diff
@@ -310,7 +311,13 @@ def for_each_mount(
         dst = os.fsencode(dst)
     if fstype:
         fstype = os.fsencode(fstype)
-    for mnt in list_for_each_entry("struct mount", ns.list.address_of_(), "mnt_list"):
+    try:
+        mounts = list_for_each_entry("struct mount", ns.list.address_of_(), "mnt_list")
+    except AttributeError:
+        mounts = rbtree_inorder_for_each_entry(
+            "struct mount", ns.mounts.address_of_(), "mnt_node"
+        )
```

Comment:
Just a couple of nits documented in `CONTRIBUTING.rst`:

1. Add a comment documenting the version divergence.
2. Put the case for the latest kernel first.

```suggestion
    # Since Linux kernel commit 2eea9ce4310d ("mounts: keep list of mounts in
    # an rbtree") (in v6.8), the mounts in a namespace are in a red-black tree.
    # Before that, they're in a list.
    try:
        mounts = rbtree_inorder_for_each_entry(
            "struct mount", ns.mounts.address_of_(), "mnt_node"
        )
    except AttributeError:
        mounts = list_for_each_entry("struct mount", ns.list.address_of_(), "mnt_list")
```

Reply:
Unfortunately that made the CI unhappy when I tried it.

Reply:
Ah, I see, `struct mnt_namespace` previously had a member named `mounts` which was the number of mounts, so the `AttributeError` doesn't trigger in the `try`. Okay, let's just add the comment.